### PR TITLE
Add issue template for pip works, uv fails cases

### PR DESCRIPTION
**Pull request by konstin**
In cases where uv doesn't work on something where pip succeeds, it often took us some back-and-forth to get the relevant information. I want to improve this with this issue template, especially asking early about their python setup. Feel free to add fields you find relevant.

### CODE REVIEW

**Review by MichaReiser — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/pip-works--uv-doesn-t.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+name: Pip works, uv doesn't
+about: For cases where `pip install` works but `uv pip install` fails
+title: ''
+labels: ''
+assignees: ''
+
+---
+
+**Description**
+<!-- If you're using a non-standard python setup, please include some context on your setup. -->
```
Comment (AlexWaygood) [inline .github/ISSUE_TEMPLATE/pip-works--uv-doesn-t.md:11]: I feel like people who know that their Python setup is non-standard will probably already put that information in an issue without this prompt, and people who don't know that their Python setup is non-standard probably won't understand what's meant by the term "non-standard" here... Maybe it's better to just ask some concrete questions here? Suggestions:
- How did you install Python? (python.org, pyenv, Windows store, conda, Linux distro, etc.)
- Please give the output of running `python -VV` on the command line

File: .github/ISSUE_TEMPLATE/pip-works--uv-doesn-t.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+**Requirements**
+<!-- Please provide a list of requirements (requirements.in or requirements.txt), ideally a minimal set that fails. -->
```
Comment (AlexWaygood) [inline .github/ISSUE_TEMPLATE/pip-works--uv-doesn-t.md:14]:
```suggestion
<!-- Please provide a list of the requirements you attempted to install with uv (requirements.in or requirements.txt), ideally a minimal set that fails. -->
```

File: .github/ISSUE_TEMPLATE/pip-works--uv-doesn-t.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+```
+
+```
+
+**pip command and output**
+<!-- The command you use to install with pip. Please make sure you are using `--no-cache-dir` to disable using cached built wheels. You can link long output as a [gist](https://gist.github.com/). -->
```
Comment (AlexWaygood) [inline .github/ISSUE_TEMPLATE/pip-works--uv-doesn-t.md:21]:
```suggestion
<!-- The working pip command. Please make sure you are using `--no-cache-dir` to disable using cached built wheels. You can link long output as a [gist](https://gist.github.com/). -->
```

File: .github/ISSUE_TEMPLATE/pip-works--uv-doesn-t.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+```shell
+pip install --no-cache-dir <your_options_here>
+```
+
+**uv command**
+<!-- The command you use to install with uv. Please make sure you are using `--no-cache`. -->
```
Comment (AlexWaygood) [inline .github/ISSUE_TEMPLATE/pip-works--uv-doesn-t.md:28]:
```suggestion
<!-- The uv command you tried. Please make sure you are using `--no-cache`. -->
```

**Review by AlexWaygood — COMMENTED** [review]
Nice. We could also consider using an issue form (e.g. see [our templates over at CPython](https://github.com/python/cpython/issues/new?assignees=&labels=type-bug&projects=&template=bug.yml)), but a template is simpler to start off with

File: .github/ISSUE_TEMPLATE/pip-works--uv-doesn-t.md (hunk 1 above)
Comment (konstin) [reply .github/ISSUE_TEMPLATE/pip-works--uv-doesn-t.md]: Those should be covered by the last two lines, i've removed this and left it as a blanket "description" header

Comment (AlexWaygood) [reply .github/ISSUE_TEMPLATE/pip-works--uv-doesn-t.md]: makese sense!

**Review by AlexWaygood — APPROVED** [review]

**Pull request closed without merging**